notebooks/time_series/apple_stock_prices.livemd

```diff
@@ -0,0 +1,264 @@
+<!-- livebook:{"file_entries":[{"name":"Apple_Stock_Prices_1981_to_2023.csv","type":"attachment"},{"name":"apple_prices.png","type":"attachment"},{"name":"normalized_apple_prices.csv","type":"attachment"},{"name":"predictions.png","type":"attachment"},{"name":"train_test_plt.png","type":"attachment"}]} -->
+
+# Time-series regression with RNNs
+
+```elixir
+Mix.install([
+  {:axon, "~> 0.6.1"},
+  {:kino_vega_lite, "~> 0.1.13"},
+  {:nx, "~> 0.7.3"},
+  {:tucan, "~> 0.3.1"},
+  {:polaris, "~> 0.1.0"},
+  {:exla, "~> 0.7.3"},
+  {:req, "~> 0.5.2"},
+  {:vega_lite, "~> 0.1.9"}
+])
+
+Nx.global_default_backend(EXLA.Backend)
+Nx.Defn.global_default_options(compiler: EXLA)
+```
+
+## Introduction
+
+In this project, we will build a predictor for a time series using an Recurrent Neural Network regressor. The prediction will be made for Apple's stock price 7 days in advance, based on a historical series.
+
+An architecture known as Long Term Short Memory (LSTM) will be used for the RNN.
+
+<!-- livebook:{"break_markdown":true} -->
+
+First, we load the historical data. Subsequently, preprocessing will be conducted to utilize the information with an RNN model. The time series data is already normalized. This helps mitigate significant numerical issues associated with how activation functions like tanh transform very large numbers (whether positive or negative), and it aids in avoiding problems with calculating derivatives.
+
+```elixir
+raw_csv =
+  "https://raw.githubusercontent.com/santiago-imelio/datasets/main/normalized_apple_prices.csv"
+  |> Req.get!()
+  |> Map.get(:body)
+
+data =
+  Regex.split(~r/\r|\n|\r\n/, raw_csv)
+  |> List.pop_at(-1)
+  |> elem(1)
+  |> Enum.map(&Float.parse/1)
+  |> Enum.map(&elem(&1, 0))
+  |> Nx.tensor()
```

Comment:
Maybe better to use Nimble CSV or Explorer to load and interact with the data a bit initially

Reply:
Added `nimble_csv` to parse the csv, although there's not much to do with the data other than visualizing, any suggestions?

Reply:
No problem then!